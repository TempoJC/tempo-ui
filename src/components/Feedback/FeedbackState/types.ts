import type { ReactNode } from "react";
import { CommonProps } from "@/types";

export type FeedbackStateProps = CommonProps & {
  /**
   * The content to show under the title and subtitle.
   */
  footer?: ReactNode;
  /**
   * The content to show over the title and subtitle.
   */
  header?: ReactNode;
  /**
   * Description text that will be shown under the title.
   */
  subtitle?: ReactNode;
  /**
   * Title of the feedback state.
   */
  title?: ReactNode;
};
