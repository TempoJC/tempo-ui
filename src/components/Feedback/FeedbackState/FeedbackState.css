.tempo-feedback-state-v2 {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  row-gap: var(--size4);
  color: var(--color-token-light-content-1);
}

.tempo-feedback-state-v2--dark {
  color: var(--color-token-dark-content-1);
}

.tempo-feedback-state-v2__message {
  display: inherit;
  flex-direction: inherit;
  align-items: inherit;
  row-gap: var(--size2);
}

.tempo-feedback-state-v2__title {
  font: var(--typography-label-l-medium-font);
  letter-spacing: var(--typography-label-l-medium-letter-spacing);
  text-transform: var(--typography-label-l-medium-text-transform);
  text-align: center;
}

.tempo-feedback-state-v2__subtitle {
  font: var(--typography-label-l-regular-font);
  letter-spacing: var(--typography-label-l-regular-letter-spacing);
  text-transform: var(--typography-label-l-regular-text-transform);
  text-align: center;
}
