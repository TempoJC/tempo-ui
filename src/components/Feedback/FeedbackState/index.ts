// packages/components-feedback/feedback-state-v2
export { default as FeedbackState } from "./FeedbackState";
export * from "./types";
