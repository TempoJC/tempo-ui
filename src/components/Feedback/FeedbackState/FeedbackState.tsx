import cn from "classnames";
import { FeedbackStateProps } from "./types";
import "./FeedbackState.css";

/**
 * A component for displaying feedback states such as loading, progress,
 * or custom messages with optional headers and footers.
 */
const FeedbackState = ({ className, testId, color = "light", title, subtitle, header, footer }: FeedbackStateProps) => (
  <div
    data-testid={testId}
    className={cn("tempo-feedback-state-v2", className, {
      "tempo-feedback-state-v2--dark": color === "dark",
    })}
  >
    {header && <div className="tempo-feedback-state-v2__header">{header}</div>}
    <div className="tempo-feedback-state-v2__message">
      {title && <div className="tempo-feedback-state-v2__title">{title}</div>}
      {subtitle && <div className="tempo-feedback-state-v2__subtitle">{subtitle}</div>}
    </div>
    {footer && <div className="tempo-feedback-state-v2__footer">{footer}</div>}
  </div>
);

export default FeedbackState;
