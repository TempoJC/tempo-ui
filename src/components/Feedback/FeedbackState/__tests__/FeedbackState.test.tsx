import { render, screen } from "@testing-library/react";
import FeedbackState from "../FeedbackState";

describe("@tempo-ui/Feedback/FeedbackState", () => {
  it("renders", () => {
    const { asFragment } = render(<FeedbackState title="Items not found" subtitle="Load another file to start" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("has title", () => {
    render(<FeedbackState title={<p>title</p>} />);
    expect(screen.getByText("title")).toBeInTheDocument();
  });

  it("has subtitle", () => {
    render(<FeedbackState subtitle={<p>support message</p>} />);
    expect(screen.getByText("support message")).toBeInTheDocument();
  });

  it("has header", () => {
    render(<FeedbackState header={<p>header</p>} />);
    expect(screen.getByText("header")).toBeInTheDocument();
  });

  it("has footer", () => {
    render(<FeedbackState footer={<p>footer</p>} />);
    expect(screen.getByText("footer")).toBeInTheDocument();
  });

  describe("testId prop", () => {
    it("renders the FeedbackState with the specified data-testid attribute", () => {
      const { getByTestId } = render(<FeedbackState testId="feedback-state" />);
      const element = getByTestId("feedback-state");
      expect(element).toBeInTheDocument();
      expect(element).toHaveAttribute("data-testid", "feedback-state");
    });
  });
});
