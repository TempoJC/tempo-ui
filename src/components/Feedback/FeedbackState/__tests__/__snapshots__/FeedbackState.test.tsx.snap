// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`@tempo-ui/Feedback/FeedbackState renders 1`] = `
<DocumentFragment>
  <div
    class="tempo-feedback-state-v2"
  >
    <div
      class="tempo-feedback-state-v2__message"
    >
      <div
        class="tempo-feedback-state-v2__title"
      >
        Items not found
      </div>
      <div
        class="tempo-feedback-state-v2__subtitle"
      >
        Load another file to start
      </div>
    </div>
  </div>
</DocumentFragment>
`;
